client/src/components/chat-box/chat-box.component.jsx

```diff
@@ -17,24 +30,54 @@ const ChatBox = () => {
   const currentUser = useSelector(state => state.user.currentUser);
   const partner = useSelector(state => state.chat.currentPartner);
 
+  const [joinRoomMessage, setJoinRoomMessage] = useState('not join room');
+  const [receivedMessage, setReceivedMessage] = useState('not received');
+  const [newRoomMessage, setNewRoomMessage] = useState('not received');
+
   const roomIds = history.map(h => h.id);
   const userId = currentUser.id;
 
   useEffect(() => {
-    if (socket.connected && roomIds.length && !joinedRoom) {
-      socket.emit('join-rooms', { roomIds, userId });
-      joinedRoom = true;
-
-      socket.on('joined-room', message =>
-        console.log(`joined room id: ${message}`)
+    if (history.length > 0) {
+      let connectRoomData = { roomIds, userId };
+      socketObject.createConnectionEvent(
+        connectRoomData,
+        setJoinRoomMessage,
+        setReceivedMessage,
+        setNewRoomMessage
       );
+    }
```

Comment:
Can you make sure that this piece of code runs only one time when the message history is done fetching and not rerun in the future?